### PR TITLE
chore(ci,deps): latest typescript, nx plugins refactoring, linting issues

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue(s)

Closes none.

## Description

Fixes broken "sap component" nx generator by bumping up typescript version. Additionally, minor refactoring was done.

## Screenshots

No visual changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,185 +1,185 @@
...
-  },
-  "lint-staged": {
-    "*": [
-      "yarn nx format:write --uncommitted",
-      "yarn eslint --fix"
-    ]
-  }
-}
\ No newline at end of file
+    "name": "fundamental-ngx",
```
Comment (g-cheishvili) [inline package.json:2]: can you run `nx format:write` again?

**Review by g-cheishvili — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (platon-rov) [reply package.json:2]: It's actually formatted :)

**Comment by InnaAtanasova** [comment]
Small question: is this PR a fix or a chore? It's only updating dependencies, concerns the "tools", not really code that concerns the users.

**Review by InnaAtanasova — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -138,6 +144,8 @@
     "jasmine-core": "4.5.0",
```
Comment (mikerodonnell89) [inline package.json:144]: Do we still need to be specifying jasmine dependencies when we are using jest?

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]

**Comment by droshev** [comment]
@platon-rov can we try to prepare this PR for merging by addressing the comments? thanks

File: package.json (hunk 2 above)
Comment (platon-rov) [reply package.json:144]: I haven't replaced jasmine everywhere but just for one project (@fundamental-ngx/nx-plugin) so we still have to keep them both.

**Pull request merged**